File: Examples/VisualBasic/Programming-Documents/Joining-Appending/RemoveSourceHeadersFooters.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Reflection
Imports System.Collections
Imports System.IO
Imports System.Text

Imports Aspose.Words.Lists
Imports Aspose.Words.Fields
Imports Aspose.Words

Public Class RemoveSourceHeadersFooters
    Public Shared Sub Run()
        ' ExStart:RemoveSourceHeadersFooters
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_JoiningAndAppending()
        Dim fileName As String = "TestFile.Destination.doc"
        Dim dstDoc As New Document(dataDir & fileName)
        Dim srcDoc As New Document(dataDir & "TestFile.Source.doc")

        ' Remove the headers and footers from each of the sections in the source document.
        For Each section As Section In srcDoc.Sections
            section.ClearHeadersFooters()
        Next section

        ' Even after the headers and footers are cleared from the source document, the "LinkToPrevious" setting 
        ' For HeadersFooters can still be set. This will cause the headers and footers to continue from the destination 
        ' Document. This should set to false to avoid this behavior.
        srcDoc.FirstSection.HeadersFooters.LinkToPrevious(False)

        dataDir = dataDir & RunExamples.GetOutputFilePath(fileName)
        dstDoc.AppendDocument(srcDoc, ImportFormatMode.KeepSourceFormatting)
        dstDoc.Save(dataDir)
        ' ExEnd:RemoveSourceHeadersFooters
        Console.WriteLine(vbNewLine & "Document appended successfully with removed source header footers." & vbNewLine & "File saved at " + dataDir)
    End Sub
End Class

File: Examples/CSharp/Programming-Documents/Joining-Appending/RemoveSourceHeadersFooters.cs
using System;
using System.IO;

using Aspose.Words;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Joining_and_Appending
{
    class RemoveSourceHeadersFooters
    {
        public static void Run()
        {
            //ExStart:RemoveSourceHeadersFooters
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_JoiningAndAppending();
            string fileName = "TestFile.Destination.doc";
            Document dstDoc = new Document(dataDir + fileName);
            Document srcDoc = new Document(dataDir + "TestFile.Source.doc");

            // Remove the headers and footers from each of the sections in the source document.
            foreach (Section section in srcDoc.Sections)
            {
                section.ClearHeadersFooters();
            }

            // Even after the headers and footers are cleared from the source document, the "LinkToPrevious" setting 
            // for HeadersFooters can still be set. This will cause the headers and footers to continue from the destination 
            // document. This should set to false to avoid this behavior.
            srcDoc.FirstSection.HeadersFooters.LinkToPrevious(false);

            dstDoc.AppendDocument(srcDoc, ImportFormatMode.KeepSourceFormatting);
            dataDir = dataDir + RunExamples.GetOutputFilePath(fileName);
            dstDoc.Save(dataDir);
            //ExEnd:RemoveSourceHeadersFooters
            Console.WriteLine("\nDocument appended successfully with source header footers removed.\nFile saved at " + dataDir);
        }
    }
}
